Loader/ComboLoader.php

```diff
@@ -0,0 +1,188 @@
+    }
+
+    /**
+     * Returns combined JS or CSS files.
+     *
+     * @param array $files File name array
+     *
+     * @return string Combined files
+     *
+     * @throws \eZ\Publish\Core\Base\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\Core\Base\Exceptions\NotFoundException
+     */
+    public function getContent(array $files)
+    {
+        $this->assertAtLeastOneFile($files);
+
+        $content = '';
+
+        foreach ($files as $file => $value) {
+            $ext = $this->findFileExtension($file);
+
+            if ($ext === 'js') {
+                if ($this->isYuiFile($file)) {
+                    $file = $this->fixUnresolvedPath($file);
+                }
+
+                $content .= $this->loadFile($file, $ext);
+            } elseif ($ext === 'css') {
+                $file = $this->fixUnresolvedPath($file);
+
+                $content .= $this->loadFile($file, $ext);
+            } else {
+                throw new NotFoundException('file type', $file);
+            }
+        }
+
+        return $content;
+    }
+
+    /**
+     * Returns content type.
+     *
+     * @param array $files File name array
+     *
+     * @return string
+     * @throws \eZ\Publish\Core\Base\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\Core\Base\Exceptions\NotFoundException
+     */
+    public function getContentType(array $files)
+    {
+        $this->assertAtLeastOneFile($files);
+
+        $file = key($files);
+
+        $ext = $this->findFileExtension($file);
+
+        if ($ext === 'js') {
+            $type = 'application/javascript';
+        } elseif ($ext === 'css') {
+            $type = 'text/css';
+        } else {
+            throw new NotFoundException('file type', $file);
+        }
+
+        return $type;
+    }
+
+    /**
+     * Asserts that files array is not empty.
+     *
+     * @param array $files
+     *
+     * @throws \eZ\Publish\Core\Base\Exceptions\InvalidArgumentException
+     */
+    private function assertAtLeastOneFile(array $files)
+    {
+        if (empty($files)) {
+            throw new InvalidArgumentException('files', 'can not be empty');
+        }
+    }
+
+    /**
+     * Returns true if JS file is YUI file.
+     *
+     * @param $filename
+     *
+     * @return bool
+     */
+    private function isYuiFile($filename)
+    {
+        return $filename[0] !== '/';
+    }
+
+    /**
+     * Fixes unresolved path.
+     *
+     * @param $filename
+     *
+     * @return string
+     */
+    private function fixUnresolvedPath($filename)
+    {
+        return $this->yuiBuildDirectory . $filename;
+    }
+
+    /**
+     * Fixes filename for JS and CSS files.
+     *
+     * Replace underscores by dots.
+     *
+     * @param $filename
+     * @param $ext
+     *
+     * @return string
+     */
+    private function fixFilename($filename, $ext)
+    {
+        return $this->webDirectory . str_replace('_' . $ext, '.' . $ext, $filename);
```

Comment:
By making `yuiBuildDirectory` relative to `webDirectory`, you wouldn't have to add the webdirectory here :wink: